sdk/data/aztables/service_client.go

```diff
@@ -189,87 +173,109 @@ type ListTablesPage struct {
+// Results fetched from the service can be evaluated by calling PageResponse on the Pager.
+// If the result is false, the value of Err() will indicate if an error occurred.
+//
+// PageResponse returns the results from the page most recently fetched from the service.
+type ListTablesPager struct {
 	client            *generated.TableClient
-	current           *generated.TableClientQueryResponse
+	current           generated.TableClientQueryResponse
 	tableQueryOptions *generated.TableClientQueryOptions
 	listOptions       *ListTablesOptions
-	err               error
+	nextTableName     *string
 }
 
 // NextPage fetches the next available page of results from the service.
 // If the fetched page contains results, the return value is true, else false.
 // Results fetched from the service can be evaulated by calling PageResponse on this Pager.
-func (p *tableQueryResponsePager) NextPage(ctx context.Context) bool {
-	if p.err != nil || (p.current != nil && p.current.XMSContinuationNextTableName == nil) {
-		return false
+func (p *ListTablesPager) NextPage(ctx context.Context) (ListTablesPage, error) {
+	req, err := p.client.QueryCreateRequest(ctx, generated.Enum1Three0, &generated.TableClientQueryOptions{
+		NextTableName: p.nextTableName,
+	}, p.listOptions.toQueryOptions())
+	if err != nil {
+		return ListTablesPage{}, err
 	}
-	var resp generated.TableClientQueryResponse
-	resp, p.err = p.client.Query(ctx, generated.Enum1Three0, p.tableQueryOptions, p.listOptions.toQueryOptions())
-	p.current = &resp
-	p.tableQueryOptions.NextTableName = resp.XMSContinuationNextTableName
-	return p.err == nil && resp.TableQueryResponse.Value != nil && len(resp.TableQueryResponse.Value) > 0
+	resp, err := p.client.Pl.Do(req)
+	if err != nil {
+		return ListTablesPage{}, err
+	}
+	if !runtime.HasStatusCode(resp, http.StatusOK) {
+		return ListTablesPage{}, runtime.NewResponseError(resp)
+	}
+
+	result, err := p.client.QueryHandleResponse(resp)
+	if err != nil {
+		return ListTablesPage{}, err
+	}
+	p.current = result
+	p.nextTableName = p.current.XMSContinuationNextTableName
+	return fromGeneratedTableQueryResponseEnvelope(p.current), nil
 }
 
-// PageResponse returns the results from the page most recently fetched from the service.
-func (p *tableQueryResponsePager) PageResponse() ListTablesPage {
-	return *fromGeneratedTableQueryResponseEnvelope(p.current)
+// More returns true if there are more pages to retrieve
+func (p *ListTablesPager) More() bool {
+	if !reflect.ValueOf(p.current).IsZero() {
+		if p.current.XMSContinuationNextTableName == nil || len(*p.current.XMSContinuationNextTableName) == 0 {
+			return false
+		}
+	}
+	return true
 }
 
-// Err returns an error value if the most recent call to NextPage was not successful, else nil.
-func (p *tableQueryResponsePager) Err() error {
-	return p.err
+// NextPageTableName returns the continuation token for the ListTablesPager
+func (p *ListTablesPager) NextPageTableName() *string {
```

Comment:
Is this necessary?  Generally speaking, if we need to expose the continuation token from a pager, we should have a unified name for the API, maybe `ContinuationToken() string` (I assume it doesn't need to be a pointer-to-string as an empty string would mean no more pages).

Reply:
Another oddity for Tables, paging entities has two continuation tokens, we can return a `(string, string)` but Jeff and I decided to go with two different methods to retrieve them. I then kept the pattern the same for the Tables pager, even though it only has a single CT.

Reply:
I see.  Are the CTs exposed in the response though?  If not, then I assume that's why they're methods on the pager.

Reply:
They are exposed in the response, we could remove these methods and just expose them from the responses.

Reply:
I'm going to remove the methods exposing continuation tokens, they are available in the response structs which is more intuitive.